### PR TITLE
Adds support for stores to the agent implementation.

### PR DESCRIPTION
**Pull request by horned-sphere**

### CODE REVIEW

File: server/swim_agent/src/agent_model/init/mod.rs [hunk 1]
```diff
@@ -225,7 +244,65 @@ where
...
+                map.insert(key, value);
+            }
+            MapMessage::Remove { mut key } => {
+                let key = init_decode(&mut key_decoder, &mut key)?;
+                map.remove(&key);
+            }
+            MapMessage::Clear => {
+                map.clear();
+            }
+            MapMessage::Take(n) => {
```
[the remark this answers is not in the document]
Comment (SirCipher) [reply server/swim_agent/src/agent_model/init/mod.rs:274]: Also, you could add a branch that short circuits if the number to remove is greater than the map length and instead it just clears the map.

[the remark this answers is not in the document]
Comment (horned-sphere) [reply server/swim_agent/src/agent_model/init/mod.rs:274]: They're `u64` as the type essentially defines the protocol (as it's not really written down anywhere). It's not really appropriate for it to be a platform dependent type. In Java, it's technically `i64` but it didn't seem sensible make it signed.

[the remark this answers is not in the document]
Comment (horned-sphere) [reply server/swim_agent/src/agent_model/init/mod.rs:274]: > Also, you could add a branch that short circuits if the number to remove is greater than the map length and instead it just clears the map.

For this part, yes.

[the remark this answers is not in the document]
Comment (SirCipher) [reply server/swim_agent/src/agent_model/init/mod.rs:274]: > They're `u64` as the type essentially defines the protocol (as it's not really written down anywhere). It's not really appropriate for it to be a platform dependent type. In Java, it's technically `i64` but it didn't seem sensible make it signed.

I've just had a look at the Java event relays and it's an integer so the panic shouldn't happen

File: server/swim_agent/src/agent_model/mod.rs [hunk 2]
```diff
@@ -384,20 +397,52 @@ where
...
+                                        let init_task = run_item_initializer(
+                                            ItemKind::Store,
+                                            name,
+                                            UplinkKind::Value,
+                                            io,
+                                            WithLengthBytesCodec::default(),
+                                            init,
+                                        );
+                                        lane_init_tasks.push(init_task.boxed());
+                                    }
```
[the remark this answers is not in the document]
Comment (horned-sphere) [reply server/swim_agent/src/agent_model/mod.rs:433]: It's deceptively difficult to split this function up without either:

1. Creating functions with truly nightmareish signatures.
2. Breaking that the future created by the async method is `Send` (if you bundle up the state into a struct you end up holding references across an await).

I've come up with a way of doing it with a macro though I'm not sure if you'll think it's an improvement.

**Review by SirCipher — APPROVED** [review]

**Pull request merged**